WordPress/Classes/Utility/WPURLRequest.m

```diff
@@ -22,6 +22,12 @@ + (NSURLRequest *)requestForAuthenticationWithURL:(NSURL *)loginUrl
     NSParameterAssert(password != nil || bearerToken != nil);
     
     NSMutableURLRequest *request = [self mutableRequestWithURL:loginUrl userAgent:userAgent];
+
+    NSString *hostname = [loginUrl host];
+    if (![hostname isEqualToString:@"wordpress.com"] && ![hostname hasSuffix:@".wordpress.com"]) {
```

Comment:
That helper takes a string, and I already have a `NSURL` that doesn't need parsing